tests/shouldwork/Signal/MergeEnable.hs

```diff
@@ -0,0 +1,36 @@
+module MergeEnable where
+
+import Clash.Prelude
+
+import qualified Clash.Explicit.Prelude as CEP
+import Clash.Explicit.Testbench
+
+topEntity
+  :: Clock System
+  -> Reset System
+  -> Enable System
+  -> Signal System Bool
+  -> Signal System (Unsigned 8)
+  -> Signal System (Unsigned 8)
+topEntity clk rst en enM =
+  let f :: HiddenClockResetEnable dom
+        => Signal dom Bool
+        -> Signal dom (Unsigned 8)
+        -> Signal dom (Unsigned 8)
+      f enM0 = mergeEnable enM0 $ register 0
+  in withClockResetEnable clk rst en (f enM)
+
+testBench
+  :: Signal System Bool
+testBench = done
+ where
+  enM = True :> True :> False :> True :> Nil
+  en  = True :> True :> True  :> True :> False :> True :> Nil
```

Comment:
For completeness maybe test all 4 possible combinations of the two enable signals.
(You're missing the case where both are False at the same time.)

Reply:
Sure, I'll add it. It might catch some weird mistake we might make in the future and it costs us virtually nothing to cover the case here.

Reply:
I don't really expected you to accidentally write an XNOR (or is it NXOR?).
But it's such a small change to test all cases, you might as well test them all.